shapeshift/web
# feat: consume opportunitiesSlice in useOtherOpportunities

## Description

Makes FOX Page APYs sane again and accidentally, paves the way for https://github.com/shapeshift/web/pull/3218 to go in - effectively a subset of it with the minimum required to wire-up FOX page properly with the opportunitiesSlice.

The guts of this is this PR:

- brings new stitching selectors, merging user/metadata in an `EarnOpportunityType` shape which we have to conform to with the current architecture, hence the large diff which includes portfolio selectors re-declarations
- adds base farming/LP opportunities constants, effectively the same as the existing foxEthSlice ones to be removed in #3218
- consumes said new selectors across the FOX page domain

## Notice




## Pull Request Type


## Issue (if applicable)

N/A, spotted in https://github.com/shapeshift/web/pull/3263

## Risk



- Loading states could be borked in FOX Page and ETH/FOX farming overview modal

## Testing



- Open FOX Page
- After opportunities metadata is loaded (after a few seconds usually), the APY should show up and it should be possible to click LP/farming opportunities
- Ensure empty wallet states for LP/farming are gracefully handled
- Ensure multi-account aggregations are gracefully handled with no regression against prod with the displayed values in FOX page

### Engineering



☝🏽 
### Operations



☝🏽 

## Screenshots (if applicable)

<img width="900" alt="image" src="https://user-images.githubusercontent.com/17035424/200950299-1817a8a3-fc8b-4376-82e1-704b9555d881.png">
